assets/dev/js/editor/regions/panel/pages/menu/menu.js

```diff
@@ -1,148 +1,94 @@
-var PanelMenuGroupView = require( 'elementor-panel/pages/menu/views/group' ),
-	PanelMenuPageView;
+import MenuPageView from 'elementor-panel/pages/menu/base';
 
-PanelMenuPageView = Marionette.CompositeView.extend( {
-	id: 'elementor-panel-page-menu',
+export default class PanelMenu extends MenuPageView {
+	initialize() {
+		this.collection = PanelMenu.getGroups();
+	}
 
-	template: '#tmpl-elementor-panel-menu',
-
-	childView: PanelMenuGroupView,
-
-	childViewContainer: '#elementor-panel-page-menu-content',
-
-	// Remove empty groups that exist for BC.
-	filter: ( child ) => {
-		return child.get( 'items' ).length;
-	},
-
-	initialize: function() {
-		this.collection = PanelMenuPageView.getGroups();
-
-		this.registerDocumentItems();
-
-		// On switch a document, re create document items.
-		elementor.once( 'document:loaded', this.registerDocumentItems );
```

Comment:
Where is the replacement? (who changes the menu on switch between documents?)

Reply:
I'll answer all the questions at once:

Since we don't want to recreate the groups on every preview load nor document switch, we create the groups statically and provide them to the menu each time it's being created.

This how the current panel menu works as well.

Reply:
On every switch document - the menu is re-created?